### DISCUSSION TITLE
Node 16 - Alpine 3.14 EOL

### DISCUSSION THREAD

**Discussion by martabal** [immich-app/immich#2406] [General]
Node 16 EOL is scheduled on September 11th, 2023. However Immich server have a dependency incompatible with node 18+ (diskusage). Moreover, server and web use alpine 3.14 which is no longer supported since 01 May 2023 (they support only security fixes). I don't know if I have to raise an issue for this.

Links : 
https://nodejs.org/en/blog/announcements/nodejs16-eol
https://www.alpinelinux.org/releases/

**Comment by bo0tzz** [comment]
#2242 should cover that.

**Reply by martabal** [reply]
#2242 seems inactive, I [succesfully updated](https://github.com/martabal/immich/commits/eol-alpine-node) alpine, node and some dependencies. Should I open a PR ?

**Reply by bo0tzz** [reply]
Yes please. alpine 3.18 just released, can you include that as well?

**Reply by martabal** [reply]
Yep, I will test it this evening, just to make sure everything work. Do you prefer node 18 or node 20 ?

**Reply by bo0tzz** [reply]
@alextran1502 thoughts?

**Reply by alextran1502** [reply]
I would go for 18 since it is LTS

**Reply by martabal** [reply]
Node has not yet released a base image with alpine 3.18. Nothing to do with it, but when I run Immich for the first time, I always get this error:
```
immich_postgres          | 2023-05-11 19:45:06.198 UTC [70] ERROR:  relation "system_config" does not exist at character 128
immich_postgres          | 2023-05-11 19:45:06.198 UTC [70] STATEMENT:  SELECT "SystemConfigEntity"."key" AS "SystemConfigEntity_key", "SystemConfigEntity"."value" AS "SystemConfigEntity_value" FROM "system_config" "SystemConfigEntity"
immich_microservices     | [Nest] 29  - 05/11/2023, 7:45:06 PM   ERROR [ExceptionHandler] relation "system_config" does not exist
```

Are you guys aware of this issue?

**Reply by jrasm91** [reply]
Yes, that's a known issue. I believe restarting microservices container and it fixes itself.
